tasks/src/tasks/select.rs

```diff
@@ -0,0 +1,154 @@
+use std::num::NonZeroU32;
+
+use imap_types::{
+    command::CommandBody,
+    flag::{Flag, FlagPerm},
+    mailbox::Mailbox,
+    response::{Code, Data, StatusBody, StatusKind},
+};
+use tracing::warn;
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug, Default)]
+pub struct SelectDataUnvalidated {
+    // required untagged responses
+    pub flags: Option<Vec<Flag<'static>>>,
+    pub exists: Option<u32>,
+    pub recent: Option<u32>,
+
+    // required OK untagged responses
+    pub unseen: Option<NonZeroU32>,
+    pub permanent_flags: Option<Vec<FlagPerm<'static>>>,
+    pub uid_next: Option<NonZeroU32>,
+    pub uid_validity: Option<NonZeroU32>,
+}
+
+impl SelectDataUnvalidated {
+    pub fn validate(self) -> Result<Self, TaskError> {
```

Comment:
This ...

```rust
impl SelectDataUnvalidated {
    fn validate(self) -> SelectData {}
}
```

... could be a pattern we expose publicly. Right now, `validate` is more like an internal helper (that could take a reference instead of moving.)

I'll add a bullet point in #191. These `Data` structures are worth discussing. Not blocking, let's make progress! :-)